lib/src/Datepicker/Datepicker.tsx

```diff
@@ -1,65 +1,364 @@
 import React from "react";
-import DatePicker from "react-date-picker";
-import { Detail } from "react-calendar";
 import "./date-picker-style.scss";
+import { padNumber } from "./formatters";
+import classnames from "classnames";
 
-const calendarAltIcon: JSX.Element = (
-    <svg name="calendar-alt" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
-        <path d="M400 64h-48V12c0-6.6-5.4-12-12-12h-8c-6.6 0-12 5.4-12 12v52H128V12c0-6.6-5.4-12-12-12h-8c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V112c0-26.5-21.5-48-48-48zM48 96h352c8.8 0 16 7.2 16 16v48H32v-48c0-8.8 7.2-16 16-16zm352 384H48c-8.8 0-16-7.2-16-16V192h384v272c0 8.8-7.2 16-16 16zM148 320h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm96 0h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm96 0h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm-96 96h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm-96 0h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm192 0h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12z" />
-    </svg>
-);
-
-export interface DatepickerProps {
-    calendarIcon?: JSX.Element;
-    className?: string;
-    clearable?: boolean;
-    clearIcon?: JSX.Element;
-    disabled?: boolean;
-    error?: string;
-    format?: string;
-    id?: string;
-    label?: string;
-    locale?: string;
-    maxDate?: Date;
-    minDate?: Date;
-    name: string;
-    onChange: (value: Date) => void;
-    placeholder?: string;
-    showLeadingZeros?: boolean;
+interface OverriddenNativeProps extends React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement> {
+    value: any;
+    min?: any;
+    max?: any;
+    onChange: any;
+}
+
+export interface DatepickerProps extends OverriddenNativeProps {
```

Comment:
Can you use the same syntax we've been using with the rest of our refactored components?

```typescript
export type DatepickerProps = JSX.IntrinsicElements["input"] & {
```